### PR TITLE
Modernize code sample

### PR DESCRIPTION
**Pull request by realmazharhussain**

### CODE REVIEW

File: assets/code-sample.md [hunk 1]
```diff
@@ -1,23 +1,28 @@
...
-    "com.example.App",
-    ApplicationFlags.FLAGS_NONE
-  );
+// ExampleApp.vala
 
-  app.activate.connect(() => {
-    var win = new Gtk.ApplicationWindow(app);
+public class ExampleApp : Gtk.Application {
+  public ExampleApp () {
+    application_id = "com.example.App";
```
Comment (lw64) [inline assets/code-sample.md:6]: Please use the `Object()` syntax here

File: assets/code-sample.md [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
+    flags = ApplicationFlags.DEFAULT_FLAGS;
+  }
+
+  public override void activate () {
+    var win = new Gtk.ApplicationWindow(this);
 
     var btn = new Gtk.Button.with_label("Hello World");
```
Comment (lw64) [inline assets/code-sample.md:13]: And leave a space before `(`'s

**Review by lw64 — CHANGES_REQUESTED** [review]
Thank you a lot! :)

File: assets/code-sample.md (hunk 1 above)
Comment (realmazharhussain) [reply assets/code-sample.md]: Done.

File: assets/code-sample.md (hunk 2 above)
Comment (realmazharhussain) [reply assets/code-sample.md]: Done

**Comment by realmazharhussain** [comment]
DEFAULT_FLAGS was introduced in a somewhat new version of GLib and FLAGS_NONE was deprecated in favor of it. That is why I used DEFAULT_FLAGS.

You are probably using a non-rolling distro that ships a somewhat older version of GLib where DEFAULT_FLAGS does not exist yet.

File: assets/code-sample.md [hunk 3]
```diff
@@ -1,23 +1,30 @@
...
+    win.present ();
+  }
+
+  public static int main (string[] args) {
+    var app = new ExampleApp ();
+    return app.run (args);
+  }
 }
 
 // Compile command (requires gtk4 package to be installed):
```
Comment (colinkiama) [inline assets/code-sample.md:28]:
```suggestion
// Compile command (requires "glib-2.0" package on version 2.74 or higher and "gtk4" package to be installed):
```

**Comment by colinkiama** [comment]
That should do it. Thanks!

**Pull request merged**